### PR TITLE
Add --auto=<option> args which replace --hot-restart and --live-reload

### PR DESCRIPTION
**Pull request by jakemac53**
Fixes https://github.com/dart-lang/webdev/issues/140

cc @kevmoo

### CODE REVIEW

File: example/web/main.dart [hunk 1]
```diff
@@ -4,10 +4,12 @@
...
 import 'dart:convert';
 import 'dart:developer';
+import 'dart:html';
 
 void main() {
   registerExtension('ext.print', (_, __) async {
     print('Hello World');
     return ServiceExtensionResponse.result(json.encode({'success': true}));
   });
+  document.body.append(SpanElement()..text = 'Hello World!!');
```
Comment (grouma) [inline example/web/main.dart]: Why is this necessary?

Comment (jakemac53) [reply example/web/main.dart]: It wasn't much of an example before :D - now at least you can play with the different --auto options and see something on the page
